surch/organization.py

```diff
@@ -19,199 +19,200 @@
-        if not os.path.isdir(cloned_repos_path):
-            os.makedirs(cloned_repos_path)
-        self.results_file_path = os.path.join(
-            results_dir, self.organization, 'results.json')
-        utils.handle_results_file(self.results_file_path, consolidate_log)
-        self.item_type = 'orgs' if organization_flag else 'users'
-        self.object_type = 'organization' if organization_flag else 'user'
-        self.cloned_repos_path = cloned_repos_path
+        results_dir = \
+            os.path.join(results_dir, 'results.json') if results_dir else None
+        self.results_file_path = results_dir or os.path.join(
+            constants.RESULTS_PATH, self.organization, 'results.json')
+        self.consolidate_log = consolidate_log
+        self.is_organization = is_organization
+        self.item_type = 'orgs' if is_organization else 'users'
+        self.cloned_repos_dir = cloned_repos_dir or os.path.join(
+            self.organization, constants.CLONED_REPOS_PATH)
         self.verbose = verbose
 
-        lgr.setLevel(logging.DEBUG if verbose else logging.INFO)
-
     @classmethod
-    def init_with_config_file(cls, config_file, verbose=False,
+    def init_with_config_file(cls,
+                              config_file,
+                              verbose=False,
                               print_result=False,
-                              remove_cloned_dir=False,
-                              organization_flag=True):
-        conf_vars = utils.read_config_file(
-            config_file=config_file,
-            print_result=print_result,
-            verbose=verbose,
-            remove_cloned_dir=remove_cloned_dir,
-            organization_flag=organization_flag)
+                              is_organization=True,
+                              remove_cloned_dir=False):
+        conf_vars = utils.read_config_file(verbose=verbose,
+                                           config_file=config_file,
+                                           print_result=print_result,
+                                           is_organization=is_organization,
+                                           remove_cloned_dir=remove_cloned_dir)
         return cls(**conf_vars)
 
-    def get_github_repo_list(
-            self,
-            url_type='clone_url',
-            repository_type='public',
-            repository_per_page=100):
-        """This method get from GitHub the git url list for cloning
-        """
-        lgr.info(
-            'Retrieving list of repositories for the {0}...'.format(
-                self.object_type))
-        auth = (self.git_user, self.git_password) if self.auth else False
-        repository_data = \
-            requests.get(
-                ORG_DETAILS_API_URL.format(
-                    self.item_type, self.organization), auth=auth)
-        if repository_data.status_code == requests.codes.NOT_FOUND:
-            lgr.error(
+    def _get_org_data(self):
```

Comment:
the function states that it gets the org data based on the input. therefore it should always return something ( even if it failed to get the actual org data ).